spec/decorators/user_decorator_spec.rb

```diff
@@ -72,6 +72,97 @@
     end
   end
 
+  describe '#pending_profile_requires_verification?' do
+    it 'returns false when no pending profile exists' do
+      profile = create(:profile)
+      user_decorator = UserDecorator.new(profile.user)
+
+      expect(user_decorator.pending_profile_requires_verification?).to eq false
+    end
+
+    it 'returns true when no active profile exists' do
+      profile = create(:profile, deactivation_reason: :verification_pending)
+      user_decorator = UserDecorator.new(profile.user)
+
+      expect(user_decorator.pending_profile_requires_verification?).to eq true
+    end
+
+    it 'returns false when active profile is newer than pending profile' do
+      user = create(:user)
+      create(:profile, deactivation_reason: :verification_pending, user: user)
+      active_profile = create(:profile, :active, :verified, user: user)
+      active_profile.update_attributes(activated_at: Time.zone.now + 1.day)
+      user_decorator = UserDecorator.new(user)
+
+      expect(user_decorator.pending_profile_requires_verification?).to eq false
+    end
+
+    it 'returns true when pending profile is newer than active profile' do
+      user = create(:user)
+      create(:profile, :active, :verified, user: user)
+      create(:profile, deactivation_reason: :verification_pending, user: user)
+      user_decorator = UserDecorator.new(user)
+
+      expect(user_decorator.pending_profile_requires_verification?).to eq true
+    end
+  end
+
+  describe '#needs_profile_phone_verification?' do
+    let(:user_decorator) { UserDecorator.new(user) }
+    let(:user) { profile.user }
+    let(:profile) do
+      create(
+        :profile,
+        deactivation_reason: :verification_pending,
+        phone_confirmed: phone_confirmed
+      )
+    end
+
+    context 'phone is not confirmed' do
+      let(:phone_confirmed) { false }
+
+      it 'returns false' do
+        expect(user_decorator.needs_profile_phone_verification?).to eq false
+      end
+    end
+
+    context 'phone is confirmed' do
+      let(:phone_confirmed) { true }
+
+      it 'returns true' do
+        expect(user_decorator.needs_profile_phone_verification?).to eq true
+      end
+    end
```

Comment:
Since the `pending_profile_requires_verification?` method is unit tested above, and since the `needs_profile_phone_verification?` method depends on that method, what do you think about stubbing the decorator to receive `pending_profile_requires_verification?` with true and false and testing those scenarios, as opposed to creating a profile with a deactivated_reason. This would make this test less brittle.